File: update_pages.py
import os
import re

directory = "c:/Users/gabri/Desktop/Portfolio Ruffini/src/app/pages/projects"
exclude_files = ["Oracle.tsx", "Tarot.tsx"]

for filename in os.listdir(directory):
    if filename.endswith(".tsx") and filename not in exclude_files:
        filepath = os.path.join(directory, filename)
        with open(filepath, 'r', encoding='utf-8') as f:
            content = f.read()

        component_name = filename[:-4] # e.g., 'CourtiBrillanti'

        # Build props interface string
        props_interface = f"""interface {component_name}Props {{\n  isMenuOpen: boolean;\n  onToggleMenu: () => void;\n}}\n\n"""

        # Replace 1: Add the Props interface before the component definition
        # Cases: 'export function X', 'export default function X'
        
        updated_content = content
        if f"export function {component_name}()" in updated_content:
            updated_content = updated_content.replace(
                f"export function {component_name}()",
                f"{props_interface}export function {component_name}({{ isMenuOpen, onToggleMenu }}: {component_name}Props)"
            )
        elif f"export default function {component_name}()" in updated_content:
            updated_content = updated_content.replace(
                f"export default function {component_name}()",
                f"{props_interface}export default function {component_name}({{ isMenuOpen, onToggleMenu }}: {component_name}Props)"
            )
        elif f"const {component_name} = ()" in updated_content:
            updated_content = updated_content.replace(
                f"const {component_name} = ()",
                f"{props_interface}const {component_name} = ({{ isMenuOpen, onToggleMenu }}: {component_name}Props)"
            )
            
        # Replace 2: Update `<Header />` to `<Header isMenuOpen={isMenuOpen} onToggleMenu={onToggleMenu} />`
        updated_content = re.sub(
            r"<Header\s*/>",
            "<Header isMenuOpen={isMenuOpen} onToggleMenu={onToggleMenu} />",
            updated_content
        )

        with open(filepath, 'w', encoding='utf-8') as f:
            f.write(updated_content)
            print(f"Updated {filename}")

File: update_pages.js
const fs = require('fs');
const path = require('path');

const directory = "c:/Users/gabri/Desktop/Portfolio Ruffini/src/app/pages/projects";
const excludeFiles = ["Oracle.tsx", "Tarot.tsx"];

fs.readdirSync(directory).forEach(filename => {
    if (filename.endsWith(".tsx") && !excludeFiles.includes(filename)) {
        const filepath = path.join(directory, filename);
        let content = fs.readFileSync(filepath, 'utf-8');

        const componentName = filename.slice(0, -4);

        const propsInterface = `interface ${componentName}Props {\n  isMenuOpen: boolean;\n  onToggleMenu: () => void;\n}\n\n`;

        if (content.includes(`export function ${componentName}()`)) {
            content = content.replace(
                `export function ${componentName}()`,
                `${propsInterface}export function ${componentName}({ isMenuOpen, onToggleMenu }: ${componentName}Props)`
            );
        } else if (content.includes(`export default function ${componentName}()`)) {
            content = content.replace(
                `export default function ${componentName}()`,
                `${propsInterface}export default function ${componentName}({ isMenuOpen, onToggleMenu }: ${componentName}Props)`
            );
        } else if (content.includes(`const ${componentName} = ()`)) {
            content = content.replace(
                `const ${componentName} = ()`,
                `${propsInterface}const ${componentName} = ({ isMenuOpen, onToggleMenu }: ${componentName}Props)`
            );
        }

        content = content.replace(
            /<Header\s*\/>/g,
            "<Header isMenuOpen={isMenuOpen} onToggleMenu={onToggleMenu} />"
        );

        fs.writeFileSync(filepath, content, 'utf-8');
        console.log(`Updated ${filename}`);
    }
});
